lib/flat_search/flats.ex

```diff
@@ -2,9 +2,28 @@ defmodule FlatSearch.Flats do
   @moduledoc """
   Flats context
   """
-
   alias FlatSearch.{Repo, Flats.Flat}
 
+  def get_flat(id) do
+    with res when not is_nil(res) <-
+           Repo.get(Flat, id) do
+      {:ok, res}
+    else
+      _error -> {:error, :not_found}
+    end
```

Comment:
with doesn't add any value to code clarity if it's one statement only, you can change this to ie. case

Reply:
Would something like that be ok? It doesn't feel right somehow :/

```
def get_flat(id) do
    case Repo.get(Flat, id) do
      nil ->
        {:error, :not_found}

      _ ->
        {:ok, Repo.get(Flat, id)}
    end
  end
  ```

Reply:
why not 
```
    case Repo.get(Flat, id) do
      nil ->
        {:error, :not_found}

      flat ->
        {:ok, flat}
    end
```

Reply:
right, don't know why I didn't think about it, thanks